app/session.js

```diff
@@ -141,12 +141,12 @@ module.exports = class Session extends EventEmitter {
   }
 
   write(data) {
-    this.pty.write(data);
+    this.pty && this.pty.write(data);
```

Comment:
I anticipated the case: A plugin overrides the constructor and prevents the pty creation (without override this `write` method). It seems legit to fail silently.

But thanks to your comment, I think this is currently pretty hard to prevent this pty creation because it occurs in constructor. Such a plugin can't call `super()` and would break other plugins that have previously decorated this Class.

I think we should isolate pty creation in a `init(options)` function that plugin would override to prevent this pty creation without breaking constructor chain.